### PR TITLE
Genericized Extremes, allows any TS and derived stat

### PR DESCRIPTION
**Pull request by thongsav-usgs**

### CODE REVIEW

File: R/extremes-data.R [hunk 1]
```diff
@@ -1,169 +1,190 @@
...
+  points <- mutate(points, 
+         value = paste(getQualifierString(points), points$value))
+  return(points)
+}
+
+flattenParam <- function(param){
+  baseParam <- strsplit(gsub("([A-Z])", " \\1", param[1]), " ")[[1]]
+  param <- paste(unique(c(baseParam, param[-1])), collapse='')
+  return(param)
+}
```
Comment (dpattermann-usgs) [inline R/extremes-data.R:190]: Did you change anything in these files? Changing line endings has made any differences hard to see.

File: inst/extremes/extremes.Rmd [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
+
+**Location:**  `r getReportMetadata(ts,'locationNumber',required=TRUE)` `r getReportMetadata(ts,'locationName',required=TRUE)`
+   
+**Period** `r getReportMetadata(ts, 'startDate')` to `r getReportMetadata(ts, 'endDate')`
+
+**Primary Series:** `r getReportMetadata(ts,'primaryLabel')`
+
+**Upchain Series:** `r getReportMetadata(ts,'upchainLabel')`
+
+**Daily Mean Series:** `r getReportMetadata(ts,'dvLabel')`
```
Comment (dpattermann-usgs) [inline inst/extremes/extremes.Rmd:13]: Are these always the mean?

File: R/extremes-data.R (hunk 1 above)
Comment (thongsav-usgs) [inline R/extremes-data.R:190]: Shoot, sorry. I forgot about my IDE doing this. Will put in another PR with line endings fixed.

File: inst/extremes/extremes.Rmd (hunk 2 above)
Comment (thongsav-usgs) [inline inst/extremes/extremes.Rmd:13]: Good catch.
